h/routes.py

```diff
@@ -112,6 +112,11 @@ def includeme(config):
     config.add_route('api.groups',
                      '/api/groups',
                      factory='h.traversal.GroupRoot')
+    config.add_route('api.group_upsert',
```

Comment:
Perhaps unsurprisingly, route ordering matters here. This route will not match correctly if it is defined after the `api.group` route (as that route would match, too, but has no handler for `PUT`).

Reply:
To protect ourselves here, we _could_ for now put an explicit `request_method='PATCH'` on the `api.group` route).

Reply:
I think the explicit request method would be more Pythonic than implicitly relying on ordering